sys-utils/nsenter.c

```diff
@@ -187,6 +191,37 @@ static int is_usable_namespace(pid_t target, const struct namespace_file *nsfile
 	return true; /* All pass */
 }
 
+static char** get_process_environment(void) {
+	char *buffer = NULL, *p = NULL;
+	size_t size = 8192, rc = 0;
+	char** envp;
+	int num;
+
+	buffer = (char*)xrealloc(buffer, size);
```

Comment:
The loop is gone, so the `xrealloc()` is just a plain `alloc()`.
It would probably best to introduce a `read_all_alloc()` helper that takes care of reading a file into an allocated buffer.

Reply:
We have ul_path_read_string(), but it reads into buf[BUFSIZ] and strdup() the result. Maybe something more elegant should be implemented.

Reply:
> We have ul_path_read_string(), but it reads into buf[BUFSIZ] and strdup() the result. Maybe something more elegant should be implemented.

Thank you for your suggestion. 
However, using `ul_path_read_string()` is not suitable as it requires a `path_cxt` which is not available in this case. Passing NULL may not be the best practice, although, it works as `ul_path_read_string()` calls `ul_path_open()` which calls `open()` if the path_cxt is NULL.